tests/knowledge_base/test_actions.py

```diff
@@ -152,3 +199,37 @@ async def test_action_run(data_file, slots, expected_slots):
             actual_msg = dispatcher.messages[0]["text"]
 
             assert actual_msg == expected_msg
+
+    # Check that temporary slot setting by action is correct.
+    if any(initial_slots.values()):
+        if (
+            initial_slots.get(SLOT_OBJECT_TYPE) is None
+            and initial_slots.get(SLOT_MENTION) is None
+        ):
+            expected_tracker_event = [
+                {
+                    "event": "slot",
+                    "timestamp": None,
+                    "name": "object_type",
+                    "value": "restaurant",
+                }
+            ]
+            actual_tracker_event = tracker.events
+
+            assert actual_tracker_event == expected_tracker_event
```

Comment:
I would rewrite this to (this way you don't need to place the expected event in a list too):
```suggestion
            assert expected_tracker_event in tracker.events
```